content/en/blog/_posts/2022-10-03-add-userns-alpha/index.md

```diff
@@ -0,0 +1,77 @@
+---
+layout: blog
+title: User namespace alpha support
+date: 2022-10-03
+slug: userns-alpha
+canonicalUrl: https://gateway-api.sigs.k8s.io/blog/2022/user-namespace-alpha-support/
+---
+
+**Authors:** Rodrigo Campos (Microsoft), Giuseppe Scrivano (Red Hat)
+
+Kubernetes v1.25 introduces the support for user namespaces.
+
+This is a major improvement for running secure workloads in
+Kubernetes.  Each pod will have access only to a limited subset of the
+available UIDs and GIDs on the system, thus adding a new security
+layer to protect from other pods running on the same system.
+
+## How does it work?
+A process running on Linux can use up to 4294967296 different UIDs and
+GIDs.
+
+User namespaces is a Linux feature that allows mapping a set of users
+in the container to different users in the host, thus restricting what
+IDs a process can effectively use.
+Furthermore, the capabilities granted in a user namespace are valid
+only in the namespace, not at the system level.
```

Comment:
Would be more accurate to say

> the capabilities granted in a new user namespace do not apply in the host initial namespaces

A good example for that would be `hostUsers: false` + `hostNetwork: true` pod with NET_ADMIN, which wouldn't be able to excercise its capability on the host networks.

Reply:
That is not a good example because you can't set  `hostUsers: false` AND `hostNetwork: true`

Reply:
I didn't know that one, I suppose I learned something!

I guess it's a limitation of the CR/k8s not the kernel right? Manually, it's pretty easy to produce with `unshare --map-root-user` then trying to alter the inherited host networks from the new user namespace.

Reply:
Yes. Some runtimes don't support it and we decided to avoid it in k8s and open the gate if needed. Lifting restrictions is something we can do in k8s, closing open doors it is not so easy.